classification_fs_all_cellprofiler_all_feat_perfect.py

```diff
@@ -0,0 +1,329 @@
+plt.xlabel('PC1')
+plt.ylabel('PC2')
+plt.legend((null, wt), ('null', 'wt'))
+plt.show()
+
+
+# In[12]:
+
+
+reducer = umap.UMAP(random_state=rnd_val)
+reducer.fit(train_feat[:,0:-1])
+
+
+# In[13]:
+
+
+null = plt.scatter(reducer.embedding_[null_idx,0],reducer.embedding_[null_idx,1], marker='x', color='r')
+wt = plt.scatter(reducer.embedding_[wt_idx,0],reducer.embedding_[wt_idx,1], marker='.', color='b')
+plt.title('Embedding of the training set by UMAP', fontsize=24)
+plt.legend((null, wt), ('null', 'wt'), loc='lower right')
+plt.show()
+
+
+# # K Cross Validation
+
+# In[14]:
+
+
+num_splits = 5 # Default number of splits
+
+def kcross_val(model, feat, splits=num_splits):
+    """
+    Parameters
+    ----------
+    model: sklearn model
+        The model to be trained and evaluated.
+    feat : Pandas Dataframe
+        The preprocessed dataframe with features and labels.
+    splits: int
+        The number of splits for k cross validation
+
+    Returns
+    -------
+    dict
+        A dictionary containing: {Best validation accuracy, Best model, Prediction of best model, Validation set labels}.
+        "Best" corresponds to the greatest validation accuracy.
+    """
+
+    cv = KFold(n_splits = splits, random_state=rnd_val, shuffle=True)
+    max_acc = 0 # Keep track of the maximum accuracy
+
+    for train_idx, val_idx in cv.split(X=feat): # iterates through index splits
+        yval = feat[val_idx,-1].astype(np.int64) # Validation labels converted to ints
+        ytrain = feat[train_idx,-1].astype(np.int64) # Training labels converted to ints
+        Xval = feat[val_idx,0:-1] # Validation features
+        Xtrain = feat[train_idx,0:-1] # Training features
+
+        model.fit(Xtrain, ytrain) # Fitting the model
+        acc = model.score(Xval, yval) # Scoring the model on the validation set
+
+        if max_acc < acc: # Find the model with the best validation accuracy
+            
+            max_acc, best_model, best_preds, val_labels = acc, model, model.predict(Xval), yval
+    
+    res = {'acc':max_acc, 'model':best_model, 'preds':best_preds, 'labels':val_labels} # Store as a dictionary
+
+    return res
+
+
+# In[15]:
+
+
+# Naive Model:
+class naive_model:
+    """
+    A naive model, which chosses a random class to generate predictions.
+    """
+    
+    def __init__(self):
+        pass
+    
+    def score(self, labels):
+        """
+        Parameters
+        ----------
+        labels
+            The genotype labels.
+
+        Returns
+        -------
+        float
+            A naive accuracy (baseline accuracy) for the given labels without considering any features.
+        """
+        naive_pred = rng.integers(low=0, high=2, size=labels.shape[0])
+        naive_acc = accuracy_score(naive_pred,labels)
+        return naive_acc
+
+
+# ## Confusion Matrix
+
+# In[16]:
+
+
+# Display Confusion matrix
+def conf_mat(model_res, mat_title='Confusion Matrix'):
```

Comment:
Same comment as above, this seems like enough functions to make into a `mlp_utils.py` file or something like that.

Reply:
Agreed, added to eval_utils instead, but I might also add this to a future pr instead.